bodhi/server/consumers/signed.py

```diff
@@ -0,0 +1,88 @@
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+The "signed handler".
+
+This module is responsible for marking builds as "signed" when they get moved
+from the pending-signing to pending-updates-testing tag by RoboSignatory.
+"""
+
+import pprint
+
+import fedmsg.consumers
+
+from pyramid.paster import get_appsettings
+from sqlalchemy import engine_from_config
+
+from bodhi.server.util import transactional_session_maker
+from bodhi.server.models import Base, Build, Release
+
+import logging
+log = logging.getLogger('bodhi')
+
+
+class SignedHandler(fedmsg.consumers.FedmsgConsumer):
+    """The Bodhi Signed Handler.
+
+    A fedmsg listener waiting for messages from koji about builds being tagged.
+
+    """
+    config_key = 'signed_handler'
+
+    def __init__(self, hub, db_factory=None, *args, **kwargs):
+        if not db_factory:
+            config_uri = '/etc/bodhi/production.ini'
+            self.settings = get_appsettings(config_uri)
+            engine = engine_from_config(self.settings, 'sqlalchemy.')
+            Base.metadata.create_all(engine)
+            self.db_factory = transactional_session_maker(engine)
```

Comment:
Code like this is repeated around Bodhi, and actually appears twice in this pull request. I think it makes sense to make this a utility function and call it from the places than need the db_factory.